gk/main.c

```diff
@@ -1114,6 +1314,12 @@ process_pkts_front(uint16_t port_front, uint16_t port_back,
 					continue;
 				}
 
+				if (update_ip_hop_count(&packet, port_front,
```

Comment:
Instead of transmitting individually an ICMP packet when `packet` must be dropped due to its TTL, `update_ip_hop_count()` could just add the ICMP packet to an array passed as a parameter. This enables the callers of `update_ip_hop_count()` to send all ICMP packet all at once. This is important because we may have many ICMP packets to send and this is critical because it's under the control of an attacker.